blocks/main_loop.py

```diff
@@ -231,8 +231,19 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                if other not in ['before_batch', 'after_batch',
```

Comment:
Now that you added the `callback` decorator, you don't need to list the callbacks here anymore. You can automatically retrieve them! You could use e.g.
```python
callbacks = [key for key, value in TrainingExtension.__dict__.items()
             if getattr(value, '_is_callback', False)]
if other not in callbacks:
    ...
```